### PR TITLE
add unit tests for OAuth2ScopeService

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
Added more unit tests for OAuth2ScopeServiceTest

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/test/resources/dbScripts/h2.sql [hunk 1]
```diff
@@ -74,3 +74,34 @@ CREATE TABLE IF NOT EXISTS  IDN_OIDC_PROPERTY (
...
+CREATE TABLE IF NOT EXISTS IDN_OAUTH2_USER_CONSENTED_SCOPES (
+    ID INTEGER NOT NULL AUTO_INCREMENT,
+    CONSENT_ID VARCHAR(255) NOT NULL,
+    TENANT_ID INTEGER NOT NULL DEFAULT -1,
+    SCOPE VARCHAR(255) NOT NULL,
+    CONSENT BOOLEAN(1) NOT NULL,
+    PRIMARY KEY (ID),
+    FOREIGN KEY (CONSENT_ID) REFERENCES IDN_OAUTH2_USER_CONSENT(CONSENT_ID) ON DELETE CASCADE,
+    UNIQUE (CONSENT_ID, SCOPE)
+);
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.oauth/src/test/resources/dbScripts/h2.sql:107]: add a new line at the end of the file

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 2]
```diff
@@ -158,4 +231,160 @@ public void testDeleteScope() throws Exception {
...
+    @Test
+    public void testRevokeUserConsentForApplication() throws Exception {
+
+        List<String> approvedScopes = new ArrayList<>(Collections.singletonList("approvedScopes"));
+        List<String> deniedScopes = new ArrayList<>(Collections.singletonList("deniedScopes"));
+        String uuid = UUID.randomUUID().toString();
+        insertAppId(uuid);
+        oAuth2ScopeService.addUserConsentForApplication("user_id", uuid, 1234, approvedScopes, deniedScopes);
+
+        oAuth2ScopeService.revokeUserConsentForApplication("user_id", uuid, 1234);
```
Comment (ChamathNS) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:318]: Is there a way to assert this?

File: components/org.wso2.carbon.identity.oauth/src/test/resources/dbScripts/h2.sql [hunk 3]
```diff
@@ -74,3 +74,34 @@ CREATE TABLE IF NOT EXISTS  IDN_OIDC_PROPERTY (
   FOREIGN KEY (CONSUMER_KEY) REFERENCES IDN_OAUTH_CONSUMER_APPS(CONSUMER_KEY) ON DELETE CASCADE
 );
 
+CREATE TABLE IF NOT EXISTS SP_APP (
+    ID                INTEGER       NOT NULL AUTO_INCREMENT,
+    TENANT_ID         INTEGER       NOT NULL,
+    APP_NAME          VARCHAR (255) NOT NULL,
+    UUID              CHAR(36),
```
Comment (ChamathNS) [inline components/org.wso2.carbon.identity.oauth/src/test/resources/dbScripts/h2.sql:81]: Formatting issue

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 4]
```diff
@@ -158,4 +231,160 @@ public void testDeleteScope() throws Exception {
...
     }
+
+    @Test
+    public void testAddUserConsentForApplication() throws Exception {
+
+        List<String> approvedScopes = new ArrayList<>(Collections.singletonList("approvedScopes"));
+        List<String> deniedScopes = new ArrayList<>(Collections.singletonList("deniedScopes"));
+        String uuid = UUID.randomUUID().toString();
+        insertAppId(uuid);
+        oAuth2ScopeService.addUserConsentForApplication("user_id", uuid, 1, approvedScopes, deniedScopes);
```
Comment (ChamathNS) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:242]: Need to assert this

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 5]
```diff
@@ -158,4 +231,160 @@ public void testDeleteScope() throws Exception {
...
+
+        List<String> approvedScopes = new ArrayList<>(Collections.singletonList("approvedScopes"));
+        List<String> deniedScopes = new ArrayList<>(Collections.singletonList("deniedScopes"));
+        String uuid = UUID.randomUUID().toString();
+        insertAppId(uuid);
+        oAuth2ScopeService.addUserConsentForApplication("user_id", uuid, 1234, approvedScopes, deniedScopes);
+
+        List<String> newApprovedScopes = new ArrayList<>(Collections.singletonList("newApprovedScopes"));
+        List<String> newDeniedScopes = new ArrayList<>(Collections.singletonList("newDeniedScopes"));
+        oAuth2ScopeService.updateUserConsentForApplication("user_id", uuid, 1234, newApprovedScopes, newDeniedScopes);
```
Comment (ChamathNS) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:288]: Assert the functionality here as well

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 6]
```diff
@@ -98,6 +109,41 @@ public void testRegisterScopeWithNoDisplayName() throws Exception {
         oAuth2ScopeService.registerScope(scope);
     }
 
+    @Test
+    public void testRegisterScopeWithExistingScopeName() throws Exception {
+
+        Scope dummyScope = new Scope(SCOPE_NAME, SCOPE_NAME, SCOPE_DESCRIPTION);
+        oAuth2ScopeService.registerScope(dummyScope);
+        try {
+            oAuth2ScopeService.registerScope(dummyScope);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:118]: Test should fail if registerScope() doesn't throw an exception

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 7]
```diff
@@ -98,6 +109,41 @@ public void testRegisterScopeWithNoDisplayName() throws Exception {
...
+        };
+    }
+
+    @Test(dataProvider = "invalidScopeNameProvider")
+    public void testRegisterWithInvalidScopeName(String scopeName, String expected) {
+
+        try {
+            Scope scope = new Scope(scopeName, "displayName", "description");
+            oAuth2ScopeService.registerScope(scope);
+        } catch (IdentityOAuth2ScopeException ex) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:142]: Test should fail if registerScope() doesn't throw an exception

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 8]
```diff
@@ -110,8 +156,35 @@ public void testGetScopesWithStartAndCount(Integer startIndex, Integer count) th
...
+    public void testGetScopesWithStartAndCountAndRequestedScopes() throws Exception {
+
+        assertNotNull(oAuth2ScopeService.getScopes(1, 2, false, "true"), "Expected a not null object");
+    }
+
+    @Test()
+    public void testIsScopeExistsWithNullName() throws IdentityOAuth2ScopeException {
+
+        try {
+            oAuth2ScopeService.isScopeExists(null);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:169]: Test should fail if registerScope() doesn't throw an exception

Check other methods which assert for exceptions

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 9]
```diff
@@ -110,8 +156,35 @@ public void testGetScopesWithStartAndCount(Integer startIndex, Integer count) th
         assertNotNull(oAuth2ScopeService.getScopes(startIndex, count), "Expected a not null object");
     }
 
+    @Test
+    public void testGetScopesWithStartAndCountAndRequestedScopes() throws Exception {
+
+        assertNotNull(oAuth2ScopeService.getScopes(1, 2, false, "true"), "Expected a not null object");
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:162]: Give a meaningful input for last argument

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java (hunk 8 above)
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:169]: Check for empty string as well

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 10]
```diff
@@ -110,8 +156,35 @@ public void testGetScopesWithStartAndCount(Integer startIndex, Integer count) th
...
+    }
+
+    @Test()
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:165]:
```suggestion
    @Test
```

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 11]
```diff
@@ -110,8 +156,35 @@ public void testGetScopesWithStartAndCount(Integer startIndex, Integer count) th
...
+    public void testIsScopeExistsWithNullName() throws IdentityOAuth2ScopeException {
+
+        try {
+            oAuth2ScopeService.isScopeExists(null);
+        } catch (IdentityException ex) {
+            assertEquals(ex.getMessage(), "Scope Name is not specified.");
+        }
+    }
+
+    @Test()
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:175]:
```suggestion
    @Test
```

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 12]
```diff
@@ -158,4 +231,193 @@ public void testDeleteScope() throws Exception {
...
+        insertAppId(uuid);
+        oAuth2ScopeService.addUserConsentForApplication("user_id", uuid, 1, approvedScopes, deniedScopes);
+
+        assertEquals(oAuth2ScopeService.getUserConsentForApp("user_id", uuid, 1).getApprovedScopes().get(0),
+                approvedScopes.get(0));
+    }
+
+    private void insertAppId(String uuid) throws Exception {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:250]: Check whether you can create the app using the DAO method

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 13]
```diff
@@ -158,4 +231,193 @@ public void testDeleteScope() throws Exception {
         oAuth2ScopeService.deleteScope(SCOPE_NAME);
         oAuth2ScopeService.getScope(SCOPE_NAME);
     }
+
+    @Test
+    public void testAddUserConsentForApplication() throws Exception {
+
+        List<String> approvedScopes = new ArrayList<>(Collections.singletonList("approvedScopes"));
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:238]:
```suggestion
        List<String> approvedScopes = new ArrayList<>(Collections.singletonList("read"));
```

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 14]
```diff
@@ -158,4 +231,193 @@ public void testDeleteScope() throws Exception {
...
+        List<String> deniedScopes = new ArrayList<>(Collections.singletonList("deniedScopes"));
+        String uuid = UUID.randomUUID().toString();
+        insertAppId(uuid);
+        oAuth2ScopeService.addUserConsentForApplication("user_id", uuid, 1, approvedScopes, deniedScopes);
+
+        assertEquals(oAuth2ScopeService.getUserConsentForApp("user_id", uuid, 1).getApprovedScopes().get(0),
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:244]: assert for denied scopes

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 15]
```diff
@@ -158,4 +231,193 @@ public void testDeleteScope() throws Exception {
...
+        }
+    }
+
+    @Test
+    public void testAddUserConsentForApplicationWithException() throws Exception {
+
+        List<String> approvedScopes = new ArrayList<>(Collections.singletonList("approvedScopes"));
+        List<String> deniedScopes = new ArrayList<>(Collections.singletonList("deniedScopes"));
+        String appId = UUID.randomUUID().toString();
+        int tenantId = 1234;
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:294]: use a constant for this and reuse wherever required

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 16]
```diff
@@ -158,4 +231,193 @@ public void testDeleteScope() throws Exception {
...
+            ps.setString(3, uuid);
+            ps.execute();
+            IdentityDatabaseUtil.commitTransaction(connection);
+        } catch (SQLException e) {
+            throw new IdentityOAuth2Exception("Error when inserting codeID", e);
+        }
+    }
+
+    @Test
+    public void testAddUserConsentForApplicationWithInvalidUserID() throws Exception {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:264]: Check null and empty

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 17]
```diff
@@ -158,4 +231,193 @@ public void testDeleteScope() throws Exception {
...
+        List<String> deniedScopes = new ArrayList<>(Collections.singletonList("deniedScopes"));
+        String uuid = UUID.randomUUID().toString();
+        insertAppId(uuid);
+        oAuth2ScopeService.addUserConsentForApplication("user_id", uuid, 1234, approvedScopes, deniedScopes);
+
+        List<String> newApprovedScopes = new ArrayList<>(Collections.singletonList("newApprovedScopes"));
+        List<String> newDeniedScopes = new ArrayList<>(Collections.singletonList("newDeniedScopes"));
+        oAuth2ScopeService.updateUserConsentForApplication("user_id", uuid, 1234, newApprovedScopes, newDeniedScopes);
+
+        assertEquals(oAuth2ScopeService.getUserConsentForApp("user_id", uuid, 1234).getApprovedScopes().size(),
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:318]: assert for denied scopes

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 18]
```diff
@@ -158,4 +231,193 @@ public void testDeleteScope() throws Exception {
...
+            oAuth2ScopeService.addUserConsentForApplication(userId, appId, tenantId, approvedScopes, deniedScopes);
+        } catch (IdentityOAuth2ScopeServerException e) {
+            String expected = String.format("Error occurred while adding user consent for "
+                    + "OAuth scopes for user : %s, application : %s and tenant Id : %d.", userId, appId, tenantId);
+            assertEquals(e.getMessage(), expected);
+        }
+    }
+
+    @Test
+    public void testUpdateUserConsentForApplication() throws Exception {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:306]: cover the case where a previously denied scope was approved later

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 19]
```diff
@@ -158,4 +231,193 @@ public void testDeleteScope() throws Exception {
...
+            assertEquals(e.getMessage(), expected);
+        }
+    }
+
+    @Test
+    public void testRevokeUserConsentForApplication() throws Exception {
+
+        List<String> approvedScopes = new ArrayList<>(Collections.singletonList("approvedScopes"));
+        List<String> deniedScopes = new ArrayList<>(Collections.singletonList("deniedScopes"));
+        String uuid = UUID.randomUUID().toString();
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:345]: appid can be something like 'invalid_app' to give a clear idea

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 20]
```diff
@@ -158,4 +231,193 @@ public void testDeleteScope() throws Exception {
...
+        String appId = UUID.randomUUID().toString();
+        insertAppId(appId);
+        int tenantId = 1234;
+        String userId = "dummyUserId";
+        oAuth2ScopeService.addUserConsentForApplication(userId, appId, tenantId, approvedScopes, deniedScopes);
+
+        OAuth2ScopeConsentResponse oAuth2ScopeConsentResponse = oAuth2ScopeService.getUserConsentForApp(userId, appId,
+                tenantId);
+        assertEquals(oAuth2ScopeConsentResponse.getAppId(), appId);
+        assertEquals(oAuth2ScopeConsentResponse.getUserId(), userId);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:368]: assert the scopes

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 21]
```diff
@@ -158,4 +231,193 @@ public void testDeleteScope() throws Exception {
...
+
+        oAuth2ScopeService.revokeUserConsentForApplication("user_id", uuid, 1234);
+
+        assertEquals(oAuth2ScopeService.getUserConsentForApp("user_id", uuid, 1234).getApprovedScopes().size(), 0);
+    }
+
+    @Test
+    public void testGetUserConsentForApp() throws Exception {
+
+        List<String> approvedScopes = new ArrayList<>(Collections.singletonList("approvedScopes"));
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:357]: use better names for scopes and use multiple scopes

**Review by mefarazath — APPROVED** [review]

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 22]
```diff
@@ -158,4 +233,264 @@ public void testDeleteScope() throws Exception {
...
+                appId, 1);
+        assertEquals(oAuth2ScopeConsentResponse.getApprovedScopes().get(0), approvedScopes.get(0));
+        assertEquals(oAuth2ScopeConsentResponse.getDeniedScopes().get(0), deniedScopes.get(0));
+    }
+
+    private void insertAppId(String uuid) throws Exception {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
+            String sql = "INSERT INTO SP_APP (TENANT_ID, APP_NAME, UUID) VALUES (?,?,?)";
+            PreparedStatement ps = connection.prepareStatement(sql);
```
Comment (ashendes) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:256]: Prepared statements and result sets (if any) should be explicitly closed as well for proper cleanup. Please add this to the `try-with-resources` statement.

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 23]
```diff
@@ -158,4 +233,264 @@ public void testDeleteScope() throws Exception {
...
+        return new Object[][]{
+                {new ArrayList<>(Arrays.asList("read", "write")),
+                        new ArrayList<>(Arrays.asList("delete")), true},
+                {new ArrayList<>(Arrays.asList("read", "write")), new ArrayList<>(), true},
+                {new ArrayList<>(), new ArrayList<>(), false}
+        };
+    }
+
+    @Test(dataProvider = "userConsentScopesProvider")
+    public void testsUserHasAnExistingConsentForApp(List<String> approvedScopes,
```
Comment (ashendes) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:447]: Inconsistent method name -> `tests`. Please fix

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java [hunk 24]
```diff
@@ -158,4 +233,264 @@ public void testDeleteScope() throws Exception {
...
+        int tenantId = 1234;
+        String userId = "dummyUserId";
+        oAuth2ScopeService.addUserConsentForApplication(userId, appId, tenantId, approvedScopes, deniedScopes);
+
+        oAuth2ScopeService.revokeUserConsents(userId, tenantId);
+
+        List<OAuth2ScopeConsentResponse> response = oAuth2ScopeService.getUserConsents(userId,
+                tenantId);
+        assertEquals(response.size(), 0);
+        }
```
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:495]: formatting issue

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java (hunk 22 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:256]: Addressed with commit [9e575a3](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1691/commits/9e575a323c0604e9985d430a5b3fee4942d86cb1)

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java (hunk 23 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:447]: Addressed with commit [9e575a3](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1691/commits/9e575a323c0604e9985d430a5b3fee4942d86cb1)

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java (hunk 24 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java:495]: Addressed with commit [9e575a3](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1691/commits/9e575a323c0604e9985d430a5b3fee4942d86cb1)

**Pull request merged**